cardinal/option_test.go

```diff
@@ -1,29 +1,12 @@
 package cardinal
 
 import (
-	"context"
-	"errors"
 	"testing"
-
-	"pkg.world.dev/world-engine/evm/x/shard/types"
-	"pkg.world.dev/world-engine/sign"
 )
 
-type DummyAdapter struct{}
-
-func (d *DummyAdapter) Submit(_ context.Context, _ *sign.Transaction, _, _ uint64) error {
-	return nil
-}
-
-func (d *DummyAdapter) QueryTransactions(_ context.Context, _ *types.QueryTransactionsRequest,
-) (*types.QueryTransactionsResponse, error) {
-	return nil, errors.New("this function should never get called")
-}
-
 func TestOptionFunctionSignatures(_ *testing.T) {
 	// This test is designed to keep API compatibility. If a compile error happens here it means a function signature to
 	// public facing functions was changed.
-	WithAdapter(&DummyAdapter{})
```

Comment:
since adapters are internal, i don't think we need to ensure it's own fn signatures change. so removing this.

Reply:
nvm just gonna change it to nil